### ISSUE TITLE
Fails silently. Doesn't show error on Promise rejection

### ISSUE THREAD

**Issue by adesege** [nestjs/passport#33]
I spent the last couple of days trying to debug why the passport module does not work. Turns out the package doesn't show thrown errors.

For example, jsonwebtoken threw an `invalid signature` error but Nestjs does not log it on the console.

Expected behaviour should be logging the error to console and throw a 500 Internal Server error to the user.

**Comment by kamilmysliwiec** [comment]
Please, provide a minimal repository which reproduces your issue.

**Comment by spali** [comment]
I tried to log the token verification errors on the server.
For this I also overwrite `handleRequest` in the JWT auth guard for graphql.
Basically I use the exact same implementation as the inherited `AuthGuard('jwt')` but with additional logging.

Use this the following auth guard on a graphql query or mutation in the resolver:
```typescript
import {
  Injectable,
  ExecutionContext,
  UnauthorizedException,
  Logger
} from '@nestjs/common';
import { AuthGuard } from '@nestjs/passport';
import { GqlExecutionContext } from '@nestjs/graphql';

/**
 * Adapted AuthGuard for GraphQL jwt authentication.
 */
@Injectable()
export class JwtAuthGuard extends AuthGuard('jwt') {
  private readonly logger = new Logger(JwtAuthGuard.name);

  getRequest(context: ExecutionContext) {
    const ctx = GqlExecutionContext.create(context);
    const request = ctx.getContext().req;
    return request;
  }

  handleRequest(err, user, info, context) {
    // valid Token: empty log line
    // invalid Token: empty log line
    this.logger.log(err, 'error');

    // valid Token: token object
    // invalid Token: false
    this.logger.log(user, 'user');

    // valid Token: empty log line
    // invalid Token: error object from passport
    this.logger.log(info, 'info');

    // valid Token: no log line at all
    // invalid Token: no log line at all
    this.logger.log(context, 'context');
    if (err || !user) {
      this.logger.error(`JWT Authentication error: ${err}`);
      throw err || new UnauthorizedException();
    }
    return user;
  }
}

```

my test output is described in the comments above the corresponding logging line.
Basically `info` contains the error which I would expect in `err`.
